prime-router/src/main/kotlin/serializers/Hl7Serializer.kt

```diff
@@ -1,19 +1,38 @@
 package gov.cdc.prime.router.serializers
 
 import ca.uhn.hl7v2.DefaultHapiContext
+import ca.uhn.hl7v2.HL7Exception
 import ca.uhn.hl7v2.model.v251.message.ORU_R01
+import ca.uhn.hl7v2.parser.CanonicalModelClassFactory
 import ca.uhn.hl7v2.util.Terser
 import gov.cdc.prime.router.Element
+import gov.cdc.prime.router.ElementAndValue
+import gov.cdc.prime.router.Mapper
 import gov.cdc.prime.router.Metadata
 import gov.cdc.prime.router.Report
+import gov.cdc.prime.router.ResultDetail
+import gov.cdc.prime.router.Schema
+import gov.cdc.prime.router.Source
 import gov.cdc.prime.router.ValueSet
+import java.io.InputStream
 import java.io.OutputStream
 import java.time.OffsetDateTime
 import java.time.ZoneId
 import java.time.format.DateTimeFormatter
 import java.util.Properties
 
 class Hl7Serializer(val metadata: Metadata) {
+    data class Hl7Mapping(
+        val mappedRows: Map<String, List<String>>,
+        val rows: List<RowResult>,
+        val errors: List<String>,
+        val warnings: List<String>,
+    )
+    data class RowResult(
+        val row: Map<String, List<String>>,
+        val errors: List<String>,
+        val warnings: List<String>,
+    )
     private val softwareVendorOrganization = "Centers for Disease Control and Prevention"
```

Comment:
Mirrors what's happening in the CSV serializer. We could potentially lift these out into their own public classes if we want.